src/Akeneo/FreeTrial/back/Infrastructure/MySql/MySqlInvitedUserRepository.php

```
@@ -24,7 +24,8 @@ public function __construct(Connection $dbalConnection)
     public function save(InvitedUser $invitedUser): void
     {
         $query = <<<SQL
-INSERT INTO akeneo_free_trial_invited_user (email, status, created_at) VALUES (:email, :status, NOW());
+INSERT INTO akeneo_free_trial_invited_user (email, status, created_at) VALUES (:email, :status, NOW())
+ON DUPLICATE KEY UPDATE status = :status;
```

We should talk about this use case. IMHO there are several ways to handle it. In that case, in the ideal scenario, the API should return an error saying that the invitation is already sent before saving it.

---

I think you misunderstood : this is only a trick to handle both create and update in one query (create = invite user via the form, update = update its status in the db). Let's talk about it tomorrow!